File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = dict()
        for i, item in enumerate(nums):
            if (target - item) in map:
                return [map[target - item],i]
            else:
                map[item] = i
        return [-1,-1]

File: Data Structures & Algorithms/two-integer-sum/submission-0.go
func twoSum(nums []int, target int) []int {
    seen := make(map[int]int)
	for index, num := range(nums) {
		if _, ok := seen[num]; ok {
			return []int{seen[num],index}
		}
		seen[target-num] = index
	}
	return []int{-1,-1}
}
